src/index.ts

```diff
@@ -204,15 +204,19 @@ const targetData: { [target: string]: TargetDataType } = {};
 export function loadDataModel(baseUrl: string, model: string, nodeId: string, target: string) : void {
   targetData[target] = { baseUrl, model, nodeId };
   const nodeURL = baseUrl + '/models/' + model + '/' + nodeId;
-  void $.getJSON(nodeURL, (data) => {
-    const root = dataToNode(data);
-    root.injectModelName(model, target);
-    setDatamodelRoot(target, root);
-
-    renderDataModels();
-  }).fail(() => {
-    throw new Error('Failed to load data model, using URL ' + nodeURL);
-  });
+  fetch(nodeURL)
+      .then(response => response.json())
+      .then(data => {
+        const root = dataToNode(data);
+        root.injectModelName(model, target);
+        setDatamodelRoot(target, root);
+
+        renderDataModels();
+      })
+      .catch(() => {
+        // TODO Alessio check here - where is it throwing? Is it intended?
```

Comment:
This is out of ignorance on my side. How could we handle this better?

Reply:
We should either return the Promise out of the function, and let the caller handle the exceptional case; or, if we only want to log it, we should log rather than throwing an exception that no code will catch.